src/core/symbology-ng/qgslinesymbollayerv2.cpp

```diff
@@ -28,6 +28,11 @@
 
 #include <cmath>
 
+inline bool evaluable( const QgsExpression * expr, const QgsSymbolV2RenderContext& context )
+{
+  return expr && ( context.feature() || !expr->referencedColumns().size() );
+}
+
```

Comment:
@vmora why not put this in QgsSymbolLayerV2 to avoid the duplicated code? I'd suggest renaming it to something like QgsSymbolLayerV2::canEvaluateExpression and also changing !expr->referencedColumns().size() to !expr->referencedColumns().isEmpty() (it's more descriptive of what is actually being tested).